server/api/utils/singletons/k8s.py

```diff
@@ -540,6 +540,65 @@ def delete_secrets(
         self.v1api.delete_namespaced_secret(secret_name, namespace)
         return mlrun.common.schemas.SecretEventActions.deleted
 
+    @raise_for_status_code
+    def create_configmap(
+        self,
+        resource: str,
+        name: str,
+        data: dict,
+        namespace: str = "",
+        labels: dict = None,
+    ):
+        namespace = self.resolve_namespace(namespace)
+
+        name = f"{resource}-{name}"
+
+        body = client.V1ConfigMap(
+            kind="ConfigMap",
+            metadata=client.V1ObjectMeta(name=name, labels=labels),
+            data=data,
+        )
+
+        try:
+            self.v1api.create_namespaced_config_map(namespace=namespace, body=body)
+        except ApiException as exc:
+            if exc.status == 409:
+                try:
+                    self.v1api.replace_namespaced_config_map(
+                        name, namespace=namespace, body=body
+                    )
+                except ApiException as exc2:
+                    logger.error(
+                        "Failed to replace k8s config map",
+                        name=name,
+                        exc=mlrun.errors.err_to_str(exc2),
+                    )
+                    raise exc2
+
+            else:
+                logger.error(
+                    "Failed to create k8s config map",
+                    name=name,
+                    exc=mlrun.errors.err_to_str(exc),
+                )
+                raise exc
+        return name
+
+    def delete_configmap(self, name: str, namespace=""):
+        namespace = self.resolve_namespace(namespace)
+
+        try:
+            self.v1api.delete_namespaced_config_map(
+                name=name,
+                namespace=namespace,
+            )
+        except ApiException as exc:
+            logger.error(
+                "Failed to delete k8s config map",
+                name=name,
+                exc=mlrun.errors.err_to_str(exc),
+            )
```

Comment:
forgot to raise?

Reply:
what will user do with that error? error in delete is almost never something you can handle